xla/service/gpu/transforms/horizontal_loop_fusion.cc

```diff
@@ -232,27 +232,6 @@ bool IsProfitableFusionCandidate(const HloInstruction& instr,
   return true;
 }
 
-// Returns whether `fusion_instr` has only row-major layouts.
```

Comment:
How does codegen handle it if we do end up with different layouts in the same fusion kernels, atleast some of the reads/writes will be uncoalesced?

Reply:
The way the check is written it doesn't actually check what would be the iteration order over physical tensors and thus is far from reasoning about coalescence. A fusion can be perfectly coalesced but rejected by this check like in the test I added or can be uncoalesced with slices / reductions / transposes but would pass through the check because it has default layouts.

Independently of that, codegen should work with any combinations of layouts, it's elementwise and doesn't enforce / rely on coalescence.